### PR TITLE
Fixes #4854 and #4462. Provides better OpenShift 4 compatibility.

### LINKED ISSUE #4854: OCP 4: syndesis-server exception: Caused by: java.net.UnknownHostException: openshift.default.svc: Name or service not known
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description
On openshift 4, I see this error in the sydnesis-server logs:
`Caused by: java.net.UnknownHostException: openshift.default.svc: Name or service not known`

Using a terminal against the server pod:
```
sh-4.2$ ping openshift.default.svc
ping: openshift.default.svc: Name or service not known
sh-4.2$ ping kubernetes.default.svc
PING kubernetes.default.svc.cluster.local (172.30.0.1) 56(84) bytes of data.
From ip-10-131-0-1.ec2.internal (10.131.0.1) icmp_seq=1 Destination Host Unreachable
From ip-10-131-0-1.ec2.internal (10.131.0.1) icmp_seq=2 Destination Host Unreachable
From ip-10-131-0-1.ec2.internal (10.131.0.1) icmp_seq=3 Destination Host Unreachable```
```

### PR DESCRIPTION
**Pull request by chirino**

### CODE REVIEW

**Comment by zregvart** [comment]
I don't think we should be using `service-ca.crt` according to this, `setCaCertFile`, is set to `ca.crt`:

https://github.com/fabric8io/kubernetes-client/blob/8b285c8ac24a5823997bd9671d1b98941719a975/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java#L389

Might be that all we need is to point the `serviceCaCertFile` to `ca.crt`, or even that we do not need to configure `setCaCertFile` as by default it will get set to `ca.crt`.

File: app/server/openshift/src/main/java/io/syndesis/server/openshift/OpenShiftConfigurationProperties.java [hunk 1]
```diff
@@ -27,13 +30,11 @@
...
 
-    public static final String SERVICE_CA_CERT_FILE = "/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt";
+    public String serviceCaCertFile = "/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt";
 
     private boolean enabled;
 
-    private String openShiftHost = "https://openshift.default.svc";
-
-    private final OpenShiftConfig openShiftClientConfig = new OpenShiftConfigBuilder().withMasterUrl(openShiftHost).withCaCertFile(SERVICE_CA_CERT_FILE).build();
+    private String openShiftHost = "https://kubernetes.default.svc";
```
Comment (heiko-braun) [inline app/server/openshift/src/main/java/io/syndesis/server/openshift/OpenShiftConfigurationProperties.java]: Does this work on 3.11? (Thinking about back porting to 1.6.x)

Comment (chirino) [reply app/server/openshift/src/main/java/io/syndesis/server/openshift/OpenShiftConfigurationProperties.java]: yes.  it works.

**Comment by chirino** [comment]
> I don't think we should be using `service-ca.crt` according to this, `setCaCertFile`, is set to `ca.crt`:
> 
> https://github.com/fabric8io/kubernetes-client/blob/8b285c8ac24a5823997bd9671d1b98941719a975/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java#L389
> 
> Might be that all we need is to point the `serviceCaCertFile` to `ca.crt`, or even that we do not need to configure `setCaCertFile` as by default it will get set to `ca.crt`.

Your likely correct.  I left it as it was as I was not testing against 3.x and was unsure if changing that would break things.

**Comment by chirino** [comment]
QE can test this change now by patching the image stream:

`oc patch -n ${TARGET_PROJECT} is syndesis-server --type='json' -p='[{"op": "replace", "path": "/spec/tags/0/from/name", "value":"quay.io/hchirino/syndesis-server:latest"}]'`

**Pull request merged**